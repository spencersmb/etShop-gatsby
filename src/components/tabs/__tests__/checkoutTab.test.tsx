import CheckoutTab from '@components/tabs/checkoutTab'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const props = {
	paymentType: 'stripe',
	handleClick: jest.fn(),
	selected: true
}
describe('Checkout Tab', () => {

	it('renders correctly', () => {
		const tree = renderer
			.create(
				<CheckoutTab {...props}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct type', () => {
		const modalRender = render(<CheckoutTab {...props}/>)
		expect(modalRender.getByTestId('tab-stripe').children[0].innerHTML).toBeDefined()
	})

	it('Should call handleClick', () => {
		const modalRender = render(<CheckoutTab {...props}/>)
		const btn = modalRender.getByTestId('tab-stripe')
		btn.click()
		expect(props.handleClick).toHaveBeenCalledTimes(1)
		expect(props.handleClick).toHaveBeenCalledWith('stripe')
	})

})
