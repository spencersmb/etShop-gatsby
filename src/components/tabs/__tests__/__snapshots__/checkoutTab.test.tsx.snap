// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Tab renders correctly 1`] = `
<li
  className="sc-htpNat fwvtES"
  data-testid="tab-stripe"
  onClick={[Function]}
  selected={true}
>
  <div
    className="stripe etRadioWrapper"
  >
    <input
      checked={true}
      id="stripe-radio"
      name="stripe-radio-group"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor="stripe-radio"
    >
      <span
        className="cc-svg"
      >
        <test-file-stub />
      </span>
      <span
        className="cc-text"
      >
        Credit Card
      </span>
    </label>
  </div>
</li>
`;
